### ISSUE TITLE
orjson 3.9.9 fails assertions with Python 3.12.0

### ISSUE THREAD

**Issue by mgorny** [ijl/orjson#436]
When running the test suite with Python 3.12 against orjson built with assertions enabled, it crashes.

```pytb
$ pip install -q pytest
$ pip install -q . -Cbuild-args=--profile=dev
$ export RUST_BACKTRACE=1
$ python -m pytest -s
========================================================= test session starts =========================================================
platform linux -- Python 3.12.0, pytest-7.4.2, pluggy-1.3.0
rootdir: /tmp/orjson-3.9.9
collected 1192 items                                                                                                                  

test/test_api.py .thread '<unnamed>' panicked at src/lib.rs:211:9:
assertion failed: ffi!(Py_REFCNT(args)) == 2
stack backtrace:
   0: rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::panicking::panic
   3: orjson::raise_loads_exception
             at ./src/lib.rs:211:9
   4: loads
             at ./src/lib.rs:294:21
   5: <unknown>
   6: _PyEval_EvalFrameDefault
   7: <unknown>
   8: _PyEval_EvalFrameDefault
   9: _PyObject_FastCallDictTstate
  10: _PyObject_Call_Prepend
  11: <unknown>
  12: _PyObject_MakeTpCall
  13: _PyEval_EvalFrameDefault
  14: _PyObject_FastCallDictTstate
  15: _PyObject_Call_Prepend
  16: <unknown>
  17: _PyObject_Call
  18: _PyEval_EvalFrameDefault
  19: _PyObject_FastCallDictTstate
  20: _PyObject_Call_Prepend
  21: <unknown>
  22: _PyObject_MakeTpCall
  23: _PyEval_EvalFrameDefault
  24: _PyObject_FastCallDictTstate
  25: _PyObject_Call_Prepend
  26: <unknown>
  27: _PyObject_MakeTpCall
  28: _PyEval_EvalFrameDefault
  29: _PyObject_FastCallDictTstate
  30: _PyObject_Call_Prepend
  31: <unknown>
  32: _PyObject_MakeTpCall
  33: _PyEval_EvalFrameDefault
  34: PyEval_EvalCode
  35: <unknown>
  36: <unknown>
  37: PyObject_Vectorcall
  38: _PyEval_EvalFrameDefault
  39: <unknown>
  40: Py_RunMain
  41: Py_BytesMain
  42: __libc_start_call_main
             at /tmp/portage/sys-libs/glibc-2.38-r6/work/glibc-2.38/csu/../sysdeps/nptl/libc_start_call_main.h:58:16
  43: __libc_start_main_impl
             at /tmp/portage/sys-libs/glibc-2.38-r6/work/glibc-2.38/csu/../csu/libc-start.c:360:3
  44: _start
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
fatal runtime error: failed to initiate panic, error 5
Fatal Python error: Aborted

Current thread 0x00007f282edb5740 (most recent call first):
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/_pytest/python_api.py", line 952 in raises
  File "/tmp/orjson-3.9.9/test/test_api.py", line 30 in test_loads_trailing_invalid
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/_pytest/python.py", line 194 in pytest_pyfunc_call
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/pluggy/_callers.py", line 77 in _multicall
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/pluggy/_manager.py", line 115 in _hookexec
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/pluggy/_hooks.py", line 493 in __call__
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/_pytest/python.py", line 1792 in runtest
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/_pytest/runner.py", line 169 in pytest_runtest_call
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/pluggy/_callers.py", line 77 in _multicall
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/pluggy/_manager.py", line 115 in _hookexec
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/pluggy/_hooks.py", line 493 in __call__
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/_pytest/runner.py", line 262 in <lambda>
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/_pytest/runner.py", line 341 in from_call
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/_pytest/runner.py", line 261 in call_runtest_hook
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/_pytest/runner.py", line 222 in call_and_report
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/_pytest/runner.py", line 133 in runtestprotocol
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/_pytest/runner.py", line 114 in pytest_runtest_protocol
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/pluggy/_callers.py", line 77 in _multicall
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/pluggy/_manager.py", line 115 in _hookexec
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/pluggy/_hooks.py", line 493 in __call__
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/_pytest/main.py", line 350 in pytest_runtestloop
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/pluggy/_callers.py", line 77 in _multicall
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/pluggy/_manager.py", line 115 in _hookexec
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/pluggy/_hooks.py", line 493 in __call__
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/_pytest/main.py", line 325 in _main
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/_pytest/main.py", line 271 in wrap_session
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/_pytest/main.py", line 318 in pytest_cmdline_main
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/pluggy/_callers.py", line 77 in _multicall
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/pluggy/_manager.py", line 115 in _hookexec
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/pluggy/_hooks.py", line 493 in __call__
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/_pytest/config/__init__.py", line 169 in main
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/_pytest/config/__init__.py", line 192 in console_main
  File "/tmp/orjson-3.9.9/.venv/lib/python3.12/site-packages/pytest/__main__.py", line 5 in <module>
  File "<frozen runpy>", line 88 in _run_code
  File "<frozen runpy>", line 198 in _run_module_as_main
Aborted (core dumped)
```

**Comment by ijl** [comment]
Thanks, fixed in [3.9.10](https://github.com/ijl/orjson/releases/tag/3.9.10).

**Comment by mgorny** [comment]
Thanks!
